arrayMethods.js

```diff
@@ -0,0 +1,488 @@
+// 1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
+//
+
+// Array.prototype.myForEach = function(callback){
+//   for(let i = 0; i< this.length; i++){
+//     callback(this[i]);
+//   }
+// };
+//
+// // test:
+// [1,2,3].myForEach((el) =>{
+//   console.log(el)
+// })
+
+//------------------------------------------------
+
+// 2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+
+// Array.prototype.myMap = function (callback){
+//   let result = [];
+//   for(let i = 0; i< this.length; i++) {
+//     result.push(callback(this[i]));
+//   }
+// return result;
+// }
+//
+// // test:
+// let mapper = [1,2,3].myMap((el) => {
+//   return el * 2
+// })
+//
+// console.log(mapper);
+
+//------------------------------------------------
+
+// 3. Write a 'myFilter` that behave's the same as `filter`.
+
+// // for loop: --------
+
+// Array.prototype.myFilter = function(callback){
+//   let newArray = []
+//   for (let i = 0; i< this.length; i++) {
+//     if(callback(this[i])){
+//       newArray.push(callback(this[i]))
+//     }
+//   }
+//   return newArray
+// }
+//
+// let filterExample = [1,2,3,4,5,6].myFilter((elem) => {
+//   if (elem % 2 === 0) {
+//     return elem
+//   }
+// })
+// console.log(filterExample);
+
+// // forEach: --------
+
+// Array.prototype.myFilter = function (callback) {
+//   let result = [];
+//   this.forEach(el => {
+//     if(callback(el)){
+//       result.push(callback (el));
+//     }
+//   })
+//   return result;
+// };
+//
+// let filterExample = ["a", "b", "c", "d", "e"].myFilter(el => {
+//   return el.toUpperCase()
+// })
+// console.log(filterExample);
+
+//------------------------------------------------
+
+//works, but:
+//stupid question: shouldn't the forloop's "return false" prevent you from getting to the prototype's "return true".
```

Comment:
Good question! The trick is, if the callback never returns false, then it never gets to the `return false` clause. Then it returns true.